mmcv/ops/csrc/common/cuda/stack_ball_query_cuda_kernel.cuh

```diff
@@ -22,54 +23,52 @@ __global__ void stack_ball_query_forward_cuda_kernel(int B, int M, float radius,
   // output:
   //      idx: (M, nsample)
   int pt_idx = blockIdx.x * blockDim.x + threadIdx.x;
-  if (pt_idx >= M)
-    return;
-
-  int bs_idx = 0, pt_cnt = new_xyz_batch_cnt[0];
-  for (int k = 1; k < B; k++) {
-    if (pt_idx < pt_cnt)
-      break;
-    pt_cnt += new_xyz_batch_cnt[k];
-    bs_idx = k;
-  }
+  CUDA_1D_KERNEL_LOOP(pt_idx, M) {
+    int bs_idx = 0;
+    for (int pt_cnt = 0; bs_idx < B; bs_idx++) {
+      pt_cnt += new_xyz_batch_cnt[bs_idx];
+      if (pt_idx < pt_cnt)
+        break;
+    }
 
-  int xyz_batch_start_idx = 0;
-  for (int k = 0; k < bs_idx; k++)
-    xyz_batch_start_idx += xyz_batch_cnt[k];
-  // for (int k = 0; k < bs_idx; k++) new_xyz_batch_start_idx +=
-  // new_xyz_batch_cnt[k];
+    int xyz_batch_start_idx = 0;
+    for (int k = 0; k < bs_idx; k++)
+      xyz_batch_start_idx += xyz_batch_cnt[k];
+    // for (int k = 0; k < bs_idx; k++) new_xyz_batch_start_idx +=
+    // new_xyz_batch_cnt[k];
 
-  new_xyz += pt_idx * 3;
-  xyz += xyz_batch_start_idx * 3;
-  idx += pt_idx * nsample;
+    const T* new_xyz_p = new_xyz + pt_idx * 3;
+    xyz += xyz_batch_start_idx * 3;
```

Comment:
Create a new variable instead of reuse `xyz` and `idx`, this might lead to invalid memory access on large input.